### DISCUSSION TITLE
Display Submit Response after Form Submit

### DISCUSSION THREAD

**Discussion by insumanth** [vueform/vueform#137] [Questions] [answered]
After form is submitted, I want to display some data based on submit response.

If the submit response is as follow

```json
{
  "created" : true,
  "reference_id": "sjZZfgSuIUyf",
  "user" : "john"
}

```

I want to use the data to display the following text in a `read only text field` or `popup` or any other way as follow

```md
Thank you for submitting the form 'John'.
Your reference id is 'sjZZfgSuIUyf'. Note this id for future refrences.
```

---------------------

A Thank you page after submitting the form with data from submit response would be ideal.

Eg: https://tally.so/help/how-to-create-a-thank-you-page

How can i acheive this in vueform.

**Comment by adamberecz** [comment, marked as answer]
First, you have to subscribe to [`@success`](https://vueform.com/reference/vueform#event-success) of `<Vueform>`. Then you can use `form$.messageBag.append('...', 'message')` to display the message, similarly as described here:
https://vueform.com/docs/validating-elements#custom-errors-and-messages

Eg:
```vue
<template>
  <Vueform @success="handleSuccess" />
</template>

<script setup>
const handleSuccess = ({ data }, form$) => {
  form$.messageBag.append(`Thank you for submitting the form '${data.user}'.<br />Your reference id is '${data. reference_id}'. Note this id for future refrences.`, 'message')
}
</script>
```
